src/handleEvent.ts

```diff
@@ -198,6 +198,12 @@ export function handleEvent<Context = unknown, Event = unknown>(
                 `${currentState} is performing a transition to ${state} - running its "-" event`
             )
 
+            // When transitioning, we have to reset all timers.
+            // Since this is pass-by-reference, we have to modify the existing array!
+            if(options.timers) {
+                options.timers.length = 0
```

Comment:
This feels like a bad idea - the garbage collector will not know that there are any timers in the array, and as such, they won't be freed like normal.